Formula/ocl-icd.rb

```diff
@@ -0,0 +1,50 @@
+class OclIcd < Formula
+  desc "OpenCL ICD loader"
+  homepage "https://github.com/OCL-dev/ocl-icd/"
+  url "https://github.com/OCL-dev/ocl-icd/archive/refs/tags/v2.3.1.tar.gz"
+  sha256 "a32b67c2d52ffbaf490be9fc18b46428ab807ab11eff7664d7ff75e06cfafd6d"
+  license "BSD-2-Clause"
+  head "https://github.com/OCL-dev/ocl-icd.git", branch: "master"
+
+  keg_only :shadowed_by_macos, "macOS provides OpenCL.framework"
+
+  depends_on "asciidoc" => :build
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "opencl-headers" => [:build, :test]
+  depends_on "xmlto" => :build
+
+  uses_from_macos "libxml2" => :build
+  uses_from_macos "libxslt" => :build
+  uses_from_macos "ruby" => :build
+
+  def install
+    ENV["XML_CATALOG_FILES"] = etc/"xml/catalog"
+    system "./bootstrap"
+    system "./configure", *std_configure_args,
+                          "--disable-silent-rules",
+                          "--enable-custom-vendordir=#{etc}/OpenCL/vendors"
```

Comment:
As I mentioned in #103141:

There is a minor question on whether `etc` is best location. The ICD files need to be updated by corresponding formulae since they contain Cellar path. I think this does happen right now when file is installed by `cmake` as opposed to `etc.install`, though I don't know if that is intended.

Reply:
Is the concern here that once the `icd` file is installed the first time, it won't be properly overridden in subsequent updates and would contain an outdated Cellar path?  I think `etc` seems like the right place to put the files but I'm not sure how we update files in there once they've been installed.  I know we don't delete files from `etc` upon uninstall, but I'm not sure if we can overwrite files there.

Reply:
> Is the concern here that once the `icd` file is installed the first time, it won't be properly overridden in subsequent updates and would contain an outdated Cellar path?

Yes, that is main worry as seen by what is inside the file:
```console
$ cat ~/.linuxbrew/etc/OpenCL/vendors/pocl.icd; echo
/home/linuxbrew/.linuxbrew/Cellar/pocl/3.0/lib/libpocl.so.2.9.0
```

---

> I think `etc` seems like the right place to put the files but I'm not sure how we update files in there once they've been installed. I know we don't delete files from `etc` upon uninstall, but I'm not sure if we can overwrite files there.

When I tried updating from PoCL 1.8 to 3.0, I did see that `cmake` overwrote the previous file. I don't know if this would behave differently if I modified `pocl.icd`

However, I am not familiar enough with brew `etc` logic to know if this is expected.

Reply:
@Homebrew/core do files in `etc` get overwritten during formula updates if they've been modified by the user?  If not we probably need to change the cellar path to the opt dir and use the unversioned soname, as that combination should still work if the file is not updated after a version update.

Reply:
`etc.install` will not overwrite files in `etc`. Practically speaking, this means that files in `etc` are never overwritten when installing from bottles.

You may find that building from source overwrites files in `etc`, though. (But that's because `make`, etc did it, and not `brew`)

Reply:
In that case, I think should `inreplace` the Cellar path with the opt dir path and the unversioned soname when installing to `etc`.

Reply:
Is `pocl.icd` meant to be user-configurable, or do you just want a persistent `OpenCL/Vendors` directory for users to stick their other `*.icd` files into?

If it's the former, we can do an `inreplace`. If it's the latter we might want to just create a symlink from `etc/"OpenCL/Vendors/pocl.icd"` to a `pocl.icd` in the appropriate Cellar/opt path.

Reply:
We should see what @cho-m thinks but my understanding is that the we just want a persistent `OpenCL/vendors` directory in `etc` for users to add `*.icd`.  The only content in `pocl.icd` is the path to the `pocl` shared library, which is not something users should be changing anyway.  So the symlinking strategy seems to make more sense to me at the moment.

Reply:
Not sure if there is a common use-case for modifying content of a formula-installed ICD file.

There are some scenarios where users may want to rename file to temporarily disable an OpenCL runtime without uninstalling (as files without `.icd` are ignored), but that should still be possible with symlinks.

---

Either approach would need to be handled in the corresponding formulae like `pocl`, `oclgrind`, ..., so current `etc` directory sounds like it will work.

Reply:
That makes sense and should be easy enough to set up in those respective formulae.  I think we've got this figured out.